### ISSUE TITLE
Group id is not expiring 

### ISSUE THREAD

**Issue by gsmtamil** [SGrondin/bottleneck#96]
As per this code snippet, the group key has to expire in Redis with TTL of 20 seconds.
After 20 sec  the key is expiring but with in 3 to 5 sec the group key is auto-creating then it repeats the same.
I am trying to delete the key using this method "group.deleteKey(key)" but it gives as undefined

I have tested in this version **2.13.0**

```js
const Bottleneck = require("bottleneck");

const express = require("express");
let app = express();

const group = new Bottleneck.Group({
  maxConcurrent: 1,
  minTime: 100,
  datastore: "redis",
  clearDatastore: false,
  timeout: 20000, // 20 seconds
  clientOptions: {
    host: "127.0.0.1",
    port: 6379
  }
});

const createLimiter = ({ id }) => {
  const limiter = group.key(id);
  limiter.on("error", err => {
    console.log(err);
  });
  return limiter;
};

const limiter = createLimiter({
  id: "GID-ABC3456RFTT456"
});

limiter
  .schedule(async () => {
    console.log("Done");
    return "Done";
  })
  .catch(error => {
    console.log(error);
  });

group.on("error", err => {
  console.log(err);
});

function clearKey() {
  let key = "b_group-key-GID-ABC3456RFTT456_settings";

  console.log("Trying to delete key");
  console.log(group.deleteKey(key)); // it gives undefined
}

setTimeout(() => {
  clearKey();
}, 30000);

app.listen(3000, function() {
  console.log("Your node is running on port 3000");
});
```

**Comment by SGrondin** [comment]
Thank you, this is an oversight on my part and an easy fix.

**Comment by gsmtamil** [comment]
Consider This scenario also: Cluster level single delete.

In this below snippet I am using two new Grop object using two group creating limiter with the same Id to simulate cluster. 
In this case, if I use deleteKey using only one Group object it is not deleting, Once TTL expires after 3 - 5 sec the key is auto-creating
If I used deleteKey method 2 times using 2 Group object it is deleting Once TTL expires.

Note: while using deleteKey, Inside group if there is no limiter, it returns undefined instant of this if it returns with promise error it would be better


`const Bottleneck = require("bottleneck");

const express = require("express");
let app = express();

const group = new Bottleneck.Group({
  maxConcurrent: 1,
  minTime: 100,
  datastore: "redis",
  clearDatastore: false,
  timeout: 20000, // 20 seconds
  clientOptions: {
    host: "127.0.0.1",
    port: 6379
  }
});

const group2 = new Bottleneck.Group({
  maxConcurrent: 1,
  minTime: 100,
  datastore: "redis",
  clearDatastore: false,
  timeout: 20000, // 20 seconds
  clientOptions: {
    host: "127.0.0.1",
    port: 6379
  }
});

const createLimiter = ({ id, maxConcurrent, minTime, clearDatastore }) => {
  const limiter = group.key(id);
  limiter.on("error", err => {
    console.log(err);
  });
  return limiter;
};

const createLimiter2 = ({ id, maxConcurrent, minTime, clearDatastore }) => {
  const limiter = group2.key(id);
  limiter.on("error", err => {
    console.log(err);
  });
  return limiter;
};

const limiter = createLimiter({
  id: "GID-ABC3456RFTT456"
});
const limiter1 = createLimiter2({
  id: "GID-ABC3456RFTT456"
});

limiter
  .schedule(async () => {
    console.log("Done");
    return "Done";
  })
  .catch(error => {
    console.log(error);
  });

group.on("error", err => {
  console.log(err);
});
group2.on("error", err => {
  console.log(err);
});

async function clearKey() {
  console.log("Trying to delete key");
  group
    .deleteKey("GID-ABC3456RFTT456")
    .then(data => {
      console.log("done delete ", data);
    })
    .catch(error => {
      console.log(error);
    });
  group2
    .deleteKey("GID-ABC3456RFTT456")
    .then(data => {
      console.log("done delete ", data);
    })
    .catch(error => {
      console.log(error);
    });
}

setTimeout(() => {
  clearKey();
}, 10000);

app.listen(3000, function() {
  console.log("Your node is running on port 3000");
});
`

**Comment by SGrondin** [comment]
Hi Tamilselvan, thank you again for your excellent bug report. The code snippets helped a lot to understand and fix the issues.

There are multiple things happening here:
1. `group.deleteKey(key)` should be called with the key name passed to `group.key(key)`, not the Redis key name. In v2.13.1, it now returns a promise that returns `true` if it successfully removed a key.
2. In v2.13.0, when calling `group.deleteKey(key)`, the key is only removed from the Group object, it will be removed from Redis after its TTL expires. The other Groups in the cluster will delete their limiter automatically once they see that it's gone from Redis. This is an oversight on my part, in v2.13.1, I've decided to change how `group.deleteKey(key)` works: it now also removes the Redis key immediately. That will speed up the time it takes for the other Groups to delete their copy.
3. You've also found a bug that was introduced in v2.10.0. The Redis key should not be recreated automatically after its TTL expires, _unless it is being used_. This was working until v2.10.0, I've fixed it in v2.13.1.

Of course I also added tests to make sure these issues won't happen again.

Wow! Thank you very much for taking the time to create this issue, it lead to *three* improvements to Groups+Clustering, the whole community will benefit. I hope you continue to use and support Bottleneck!

**Comment by SGrondin** [comment]
Yes, I saw this edge case while I was making the other changes, but it was potentially more work than the other 3 combined so I decided to wait. It will be added very soon.
